src/gt4py/next/iterator/ir_utils/ir_makers.py

```diff
@@ -498,6 +498,28 @@ def _impl(*its: itir.Expr) -> itir.FunCall:
     return _impl
 
 
+def cast_as_fieldop(type_: str, domain: Optional[itir.FunCall] = None):
```

Comment:
I am not 100% sure if this is the right place.
If it is the right place (in the sense of lack of alternatives) I would suggest that you change the word "Promote" from the description (the name of the functionname is okay).
You see for me a promotion is not a maker but something that changes something else that is already there, which is as far as I can tell from the example, not the case.